src/bmp.c

```diff
@@ -1,6 +1,48 @@
 #include <stdio.h>
+#include <stdlib.h>
+
 #include "bmp.h"
 
-void hello(){
-    printf("Hello, Bruno!\n");
+int round4(int x) {
+    return x % 4 == 0 ? x : x - x % 4 + 4;
+}
+
+void write_bmp(char *filename, char rgb[], int length, int width) {
+    int height = (length / 3) / width;
+
+    // Pad the width of the destination to a multiple of 4
+    int padded_width = round4(width * 3);
+    
+    int bitmap_size = height * padded_width * 3;
+    char *bitmap = (char *) malloc(bitmap_size * sizeof(char));
+    for (int i = 0; i < bitmap_size; i++) bitmap[i] = 0x00;

```

Comment:
I personally find same-line loops difficult to read.
```
for (int i = 0; i < bitmap_size; i++) 
    bitmap[i] = 0x00; // works much best for me
```

Also you are iterating some memory to set it to 0. That is usually done with the memset function: https://man7.org/linux/man-pages/man3/memset.3.html
```
size_t mem_size = bitmap_size * sizeof(char);
char* bitmap = malloc(mem_size);
memset(bitmap, 0, mem_size);
```